hedera-node/hedera-app/src/xtest/java/README.md

```diff
@@ -0,0 +1,47 @@
+# XTests
+
+## Introduction
+XTests are intended to fill the gap between unit tests and integration tests. 
+They are designed to test the functionality of a services module in its entirety, as well as 
+interactions between module.
+
+## Goals and Benefits
+- Easier to write, read and understand, and to debug into than the HAPI suite BDDs because they just exercise the service (or services directly) through its API instead of being an end-to-end test where a HAPI transaction goes in and a record stream comes out, with all kinds of concurrency and asynchronicity happening in between, and the measurement of the effect being rather indirect.
+- At the same time making it easier to test edge cases because you can hand-craft test data using the actual data structures taken by the API, and can thus drive the system into states it can't normally get to by using only valid HAPI transactions.
+- A more complete test of the service-under-test than (our current) unit tests because mocking is extremely discouraged. So you're actually testing the real code instead of hard-wiring in to the test a bunch of assumptions of how the real code is supposed to work.
```

Comment:
to clarify: "because mocking is extremely discouraged" -> "because mocking in XTests is extremely discouraged" ? (just in case anyone hasn't figured out it is the _unit tests_ that are full of mocks)